app/src/main/java/com/github/sdpteam15/polyevents/database/FirestoreDatabaseProvider.kt

```diff
@@ -1,192 +1,81 @@
+import com.github.sdpteam15.polyevents.database.objects.*
 import com.github.sdpteam15.polyevents.database.observe.Observable
 import com.github.sdpteam15.polyevents.database.observe.ObservableList
-import com.github.sdpteam15.polyevents.model.*
-import com.github.sdpteam15.polyevents.util.*
-import com.google.android.gms.maps.model.LatLng
+import com.github.sdpteam15.polyevents.login.UserLogin
+import com.github.sdpteam15.polyevents.model.UserEntity
+import com.github.sdpteam15.polyevents.model.UserProfile
+import com.github.sdpteam15.polyevents.util.AdapterInterface
+import com.github.sdpteam15.polyevents.util.UserAdapter
 import com.google.android.gms.tasks.OnFailureListener
 import com.google.android.gms.tasks.OnSuccessListener
 import com.google.android.gms.tasks.Task
 import com.google.firebase.auth.FirebaseAuth
-import com.google.firebase.firestore.*
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.QuerySnapshot
 import com.google.firebase.firestore.ktx.firestore
 import com.google.firebase.ktx.Firebase
 
 object FirestoreDatabaseProvider : DatabaseInterface {
+    @SuppressLint("StaticFieldLeak")
     var firestore: FirebaseFirestore? = null
         get() = field ?: Firebase.firestore
 
-    /**
-     * Map used in the firstConnection method. It's public to be able to use it in tests
-     */
-    var firstConnectionUser: UserEntity = UserEntity(uid = "DEFAULT")
-
-    override val currentUser: UserEntity?
-        get() =
-            if (FirebaseAuth.getInstance().currentUser != null) {
-                FirebaseUserAdapter.toUser(FirebaseAuth.getInstance().currentUser!!)
+    override var currentProfile: UserProfile? = null
+        get() = null
+
+    override var itemDatabase: ItemDatabaseInterface? = null
+        get() = field ?: ItemDatabaseFirestore
+    override var zoneDatabase: ZoneDatabaseInterface? = null
+        get() = field ?: ZoneDatabaseFirestore
+    override var userDatabase: UserDatabaseInterface? = null
+        get() = field ?: UserDatabaseFirestore
+    override var heatmapDatabase: HeatmapDatabaseInterface? = null
+        get() = field ?: HeatmapDatabaseFirestore
+    override var eventDatabase: EventDatabaseInterface? = null
+        get() = field ?: EventDatabaseFirestore
+    override var materialRequestDatabase: MaterialRequestDatabaseInterface? = null
+        get() = field ?: MaterialRequestDatabaseFirestore
+
+    override val currentUserObservable = Observable<UserEntity>()
+    private var loadSuccess = false
+    override var currentUser: UserEntity?
+        get() {
+            if(UserLogin.currentUserLogin.isConnected()){
+                if(!loadSuccess) {
+                    currentUserObservable.postValue(UserLogin.currentUserLogin.getCurrentUser()!!,this)
+                    firestore!!.collection(USER_COLLECTION.toString())
```

Comment:
`USER_COLLECTION.value` not `USER_COLLECTION.toString()`

Reply:
Yes, at first, I was using .toString() then I thought it was better to use .value. Thanks for poiting this out, I'll change it